train.py

```diff
@@ -0,0 +1,122 @@
+import os
+import time
+
+import hydra
+import torch
+import psutil
+import imageio
+import numpy as np
+from omegaconf import DictConfig
+from torchvision.utils import save_image
+
+import wandb
+from utils import get_interpolations
+from data_loader import get_data_loaders
+from models.autoencoder import Autoencoder
+from train_utils import test_epoch, train_epoch, prepare_loss_function
+
+
+def initialize_model(cfg: DictConfig, train_loader, test_loader, input_size):
+    if cfg.model.type == "AE":
+        model = Autoencoder(cfg, train_loader=train_loader, test_loader=test_loader, input_size=input_size)
+    else:
+        raise NotImplementedError("Model type not implemented")
+    return model
+
+
+def train(cfg: DictConfig, autoencoder: Autoencoder, loss_function: torch.nn.Module):
+    os.makedirs(cfg.logger.results_path, exist_ok=True)
+    os.makedirs(cfg.logger.checkpoint_path, exist_ok=True)
+
+    torch.manual_seed(cfg.system.seed)
+
+    train_autoencoder(cfg, autoencoder, loss_function)
+    draw_interpolation_grid(cfg, autoencoder)
```

Comment:
```suggestion
```

I think drawing the interpolation shouldn't be a part of training (unless you're doing it every epoch) - so I would move this call to `main()`.